geom/src/lib.rs

```diff
@@ -36,5 +36,21 @@ pub use crate::time::Time;
 pub const EPSILON_DIST: Distance = Distance::const_meters(0.01);
 
 pub fn trim_f64(x: f64) -> f64 {
-    (x * 10_000.0).round() / 10_000.0
+    let v1 = compress(x);
+    let v2 = compress(v1);
+    if v1 != v2 {
+        panic!("trim_f64({}) = {} v1, {} v2", x, v1, v2);
```

Comment:
Example failure: `trim_f64(648.2998884804397) = 648.2999267578125 v1, 648.2999877929688 v2`

@michaelkirk, any experience from georust here? In geom, we're already trimming all numbers to 4 decimal places to get deterministic serialization, and because we don't really need more precision in `Distance`, `Duration`, etc. Now I want to take advantage of this and get smaller files. But I'm attempting the round-tripping here, and seeing problems. I don't have a great understanding of the IEEE floating pt standards, but I know the amount of precision supported is kind of dynamic.

Some general ideas:
- Natively work with f32's in geom instead. I've attempted that cutover a few times and hit some problem or another in map importing, but it's been a while; the pipeline has been vastly robustified since then, so maybe another try is worth it.
- Use one of the many fixed precision crates?

Reply:
I have zero context, so I might be missing something, but my first thought is that doing base10 rounding on a number stored in binary seems fraught with peril.

I would expect something like:
```
let some_f32 = some_f64 as f32
```

Reply:
More specifically, could we instead round to the nearest 2^-14 (~0.000061) instead of the nearest 0.0001?